src/main/scala/jtag/JtagTap.scala

```diff
@@ -104,23 +112,18 @@ class JtagTapController(irLength: Int, initialInstruction: BigInt)(implicit val
   irChain.io.chainIn.update := currState === JtagState.UpdateIR.U
   irChain.io.capture.bits := "b01".U
 
-  val updateInstruction = Wire(Bool())
-
-  val nextActiveInstruction = Wire(UInt(irLength.W))
-  val activeInstruction = NegEdgeReg(clock, nextActiveInstruction, initialInstruction.U, updateInstruction, name = Some("irReg"))
-   // 7.2.1d active instruction output latches on TCK falling edge
-
-  when (currState === JtagState.UpdateIR.U) {
-    nextActiveInstruction := irChain.io.update.bits
-    updateInstruction := true.B
-  } .otherwise {
-    // Needed when using chisel3._ (See #1160)
-    nextActiveInstruction := DontCare
-    updateInstruction := false.B
+  withClockAndReset(clock_falling, io.control.jtag_reset) {
+    val activeInstruction = RegInit(initialInstruction.U(irLength.W))
+    when (tapIsInTestLogicReset) {
+      activeInstruction := initialInstruction.U
+    }.elsewhen (currState === JtagState.UpdateIR.U) {
+      activeInstruction := irChain.io.update.bits
+    }
+    io.output.instruction := activeInstruction
   }
-  io.output.instruction := activeInstruction
 
-  io.output.reset := currState === JtagState.TestLogicReset.U
+  tapIsInTestLogicReset := currState === JtagState.TestLogicReset.U
+  io.output.reset := tapIsInTestLogicReset
```

Comment:
can we rename this output as well to `tapIsInTestLogicReset` for clarity?

Reply:
done